interactive/shell.py

```diff
@@ -145,3 +146,39 @@ def show(self, test_set):
             self.tt._tprint(items)
         else:
             self.err()
+
+    @line_magic
+    def cache(self, line, ident='pytest/interactive'):
+        """
+            self.err("You must provide a key:"
+                     " `cache <local name> as <key>`")
+        """
+        cachedict = self.tt.get_cache_dict()
+        if line:
+            name, _, cache_key = line.partition(' as ')
+
+            # create a new entry
+            if cache_key:
+                testset = self.ns_eval(name)
+                self.tt.set_cache_items(cache_key, testset)
+
+                # update the local shell's ns
+                if testset:
+                    self.shell.user_ns[cache_key] = testset
+                return
+
+            # delete an entry
+            name, delim, cache_key = line.partition('del ')
+            if delim:
+                self.tt.set_cache_items(cache_key, None)  # delete
+                self.shell.user_ns.pop(cache_key)
+                self.tr.write(
+                    "Deleted cache entry for '{}'\n".format(cache_key))
+                return
```

Comment:
Tokenize the string first, consume the tokens one by one.

```python
tokens = line.split()
assert tokens[0] == 'cache'

if tokens[1] == 'del':
    assert len(tokens) == 3  # cache del foo
    name = tokens[2]
    # ...
else:
    assert len(tokens) == 4  # cache foo as ...
    assert token[2] == 'as'
    name = tokens[1]
    # ...
```

Which will also more correctly handle the case of putting too many whitespaces. Since you don't strip after you partition, i wonder how the following inputs would be handled:

- `cache foo  as bar` <- notice there is two spaces
- `cache some random string as there's more to come`
- `cache as as as`